File: February 2021/20.02.2021/suleyman.java
class Solution {
    public int pivotIndex(int[] nums) {
        int total = 0;
        for (int num : nums) {
            total+= num;
        }
        
        
        int currentTotal = 0;
        for (int i = 0; i < nums.length; i++) {
            
            if (currentTotal == (total - nums[i] - currentTotal)) {
                return i;
            }
            currentTotal += nums[i];
        }
        return -1;
    }
}

File: July 2021/06.07.2021/suleyman.py
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        _map = {}
        
        for num in arr:
            if num in _map:
                _map[num] += 1
            else:
                _map[num] = 1
        
        total = 0
        counter = 0
        for key in sorted(_map, key=_map.get, reverse=True):
            total += _map[key]
            counter += 1
            if total >= (len(arr) / 2):
                return counter
